cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/utils/CassandraMessageUtils.java

```diff
@@ -58,7 +58,12 @@ public static Select prepareSelect(String keyspace, String tableName, UUID insta
 							id.getStreamName(), 
 							id.getDirection().getLabel(),
 							id.getIndex(),
-							readAttrs));
+							readAttrs))
+				.thenApply(batch ->{
+					if (batch == null || batch.getLastMessageIndex() >= id.getIndex())

```

Comment:
Since we are searching for a batch, it is enough to check if `batch.getMessageIndex() == id.getIndex()`.

Reply:
Oh, I've found that getMessageBatch() method should return batch that contains message with given ID and batch itself can have different ID.
So, your change is good.

Reply:
In case the "messages" table is empty or a non-existing stream is used in id, we will get NPE. Also, why should we check batch.getMessageIndex() in condition? This way we can only find the first message in the batch.